### ISSUE TITLE
[Feature Request] Programmatic Export/Save of visConfig JSON

### ISSUE THREAD

**Issue by noamanemobidata** [Kanaries/GWalkR#38]
It would be helpful to add a function that allows users to programmatically export or save the visConfig JSON.

**Comment by bruceyyu** [comment]
This is not easy to implement. As we are using htmlwidget in R to render GWalkR's core HTML app, we are unable to get the real-time update of the visConfig from the HTML in R environment due to the lack of bi-directional communication.

**Comment by noamanemobidata** [comment]
A common use case involves saving visualizations (the json config) so that they can be easily accessed and reviewed later, without the need to recreate them each time.

**Comment by bruceyyu** [comment]
<img width="362" alt="image" src="https://github.com/user-attachments/assets/a9f7c08c-8d40-4544-a23a-44e6827a7ec7">

@noamanemobidata actually you can press this button to download the JSON config file

**Comment by noamanemobidata** [comment]
Yes, but we want to do this programmatically :  save and restore visualizations, especially in interactive applications such as Shiny. The goal is to enable users to save their current visualization as a JSON (e.g., in a database or S3 storage) with a "Save Viz" button, making it easy to restore the visualization state later.
